runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -316,11 +316,7 @@ class ExecutionContext final {
         memory_man_ptr = std::make_unique<MemoryManager>();
     }
 
-    ~ExecutionContext()
-    {
-        memory_man_ptr.reset(nullptr);
-        py_guard.reset(nullptr);
-    }
```

Comment:
Just curious about this one, did it turn out to be unnecessary? :D

Reply:
yeah, they are not necessary.. but you may see them in some old c++ programming style.